examples/3d/split_screen.rs

```diff
@@ -83,26 +81,25 @@ fn set_camera_viewports(
     mut resize_events: EventReader<WindowResized>,
     mut left_camera: Query<&mut Camera, (With<LeftCamera>, Without<RightCamera>)>,
     mut right_camera: Query<&mut Camera, With<RightCamera>>,
+    mut setup_done: Local<bool>,
 ) {
     // We need to dynamically resize the camera's viewports whenever the window size changes
     // so then each camera always takes up half the screen.
-    // A resize_event is sent when the window is first created, allowing us to reuse this system for initial setup.
-    for resize_event in resize_events.iter() {
-        if resize_event.id == WindowId::primary() {
-            let window = windows.primary();
-            let mut left_camera = left_camera.single_mut();
-            left_camera.viewport = Some(Viewport {
-                physical_position: UVec2::new(0, 0),
-                physical_size: UVec2::new(window.physical_width() / 2, window.physical_height()),
-                ..default()
-            });
+    if resize_events.iter().next().is_some() || !*setup_done {
```

Comment:
Alternatively, maybe we should send an initial WindowResized event on platforms that don't do that automatically? That way userspace doesn't need this complexity and we close a "bevy behavior consistency gap"?

Reply:
I can only test on macOS and wasm, where this event isn't sent. For me reading it in that comment was a first discovery 😄 
Could someone confirm for Windows, X11 and Wayland?

Reply:
I'm now sending the event on window creation. it shouldn't be an issue anyway if there are two events instead of one on some platforms

Reply:
Yeah I can test those platforms if nobody else beats me to it :)

Reply:
X11 has a duplicate event now (and it doesn't on main). Gonna try wayland next (both xwayland and native wayland).

Reply:
xwayland also has the duplicate event. native wayland is unclear because it crashed on startup for me (the default swap chain format wasnt compatible ... i queried the surface for the preferred format, but even after setting that, the device times out). Nvidia proprietary drivers + wayland are notoriously bad.

Windows also now has a duplicate event. (it actually produces 3 resize events on main, one with a higher dpi (2x), and two with the "actual" resolution). With this PR it produces 4 resize events.

So its safe to say that xwayland, x11, and windows don't need the extra event. Native wayland is unclear. I think we'll need someone else to test that. But "native wayland" is not the default behavior, and some platforms already produce duplicate events, so I don't think adding a potential duplicate to native wayland is particularly risky.

Reply:
I added a gate for windows and x11, not sure how to gate for xwayland but not native wayland.

Anyway as you mention, this kind of event should not be expected to be unique so it should be OK